tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/types/package.scala

```diff
@@ -165,23 +38,13 @@ package object types {
      */
     def fromType1(tt: TlaType1): CellT = {
```

Comment:
I've a general clarifying question: iiuc, it looks like the only way to construct a `CellT` is via a `TlaTyp1` (`fromTypeTag` is just a wrapper around getting the `TlaType1` from a type tag). So we not just define `CellT` as a case class that wraps a `TlaType1` directly, rather than defining a case class of the trait?

Reply:
A spitball idea: would it make any sense to define `case class CellT[T <: TlaType1](tt: T)` so that we'd be able to take advantage of static analysis when we'd narrowed down to a specific type?

Reply:
`CellT` has several case classes that do not fit into `TlaType1` directly. These case classes with disappear in the future refactorings. Ultimately, we should use `TlaType1` only.